### PR TITLE
Add Bank Identifier Code assert

### PR DESCRIPTION
**Pull request by nunofgs**
Adds an assert to test whether a value is a valid Bank Identifier Code (*BIC*) as defined in the [ISO-9362](http://www.iso.org/iso/home/store/catalogue_tc/catalogue_detail.htm?csnumber=60390) standard.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ Some asserts may require additional peer dependencies. Note that from [npm 3](ht
 
 The following set of extra asserts are provided by this package:
 
+* BankIdentifierCode
```
Comment (ruimarinho) [inline README.md]: `BankIdentifierCode (BIC)`?

File: README.md [hunk 2]
```diff
@@ -47,6 +48,10 @@ The following set of extra asserts are provided by this package:
 * UsState (requires `provinces`)
 * Uuid
 
+### BankIdentifierCode
+
+Tests if the value is a valid Bank Identifier Code (BIC).
```
Comment (ruimarinho) [inline README.md]: Do you have any link to provide here?

File: src/asserts/bank-identifier-code-assert.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+   * Class name.
+   */
+
+  this.__class__ = 'BankIdentifierCode';
+
+  /**
+   * Validation algorithm.
+   */
+
+  this.validate = (value) => {
```
Comment (ruimarinho) [inline src/asserts/bank-identifier-code-assert.js]: Remove parenthesis please.

File: src/asserts/bank-identifier-code-assert.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    if (typeof value !== 'string') {
+      throw new Violation(this, value, { value: Validator.errorCode.must_be_a_string });
+    }
+
+    if (bic.test(value) !== true) {
```
Comment (ruimarinho) [inline src/asserts/bank-identifier-code-assert.js]: `!bic.test(value)`?

File: test/asserts/bank-identifier-code-assert_test.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+const Assert = BaseAssert.extend({
+  BankIdentifierCode: BankIdentifierCodeAssert
+});
+
+describe('BankIdentifierCodeAssert', () => {
+  it('should throw an error if the input value is not a string', () => {
+    const choices = [[], {}];
+
+    choices.forEach((choice) => {
```
Comment (ruimarinho) [inline test/asserts/bank-identifier-code-assert_test.js]: Remove parenthesis.

**Comment by ruimarinho** [comment]
Other than the small comments, LGTM. Thanks!

**Comment by nunofgs** [comment]
@ruimarinho Updated.

File: test/asserts/bank-identifier-code-assert_test.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+/**
+ * Extend `Assert` with `BankIdentifierCodeAssert`.
+ */
+
+const Assert = BaseAssert.extend({
+  BankIdentifierCode: BankIdentifierCodeAssert
+});
+
+describe('BankIdentifierCodeAssert', () => {
```
Comment (ruimarinho) [inline test/asserts/bank-identifier-code-assert_test.js]: Missing:

```
/**
 * Test `BankIdentifierCodeAssert `.
 */
```

**Pull request merged**